src/test/java/org/broadinstitute/hellbender/tools/GenomicsDBImportIntegrationTest.java

```diff
@@ -26,68 +29,150 @@
+    public void testGenomicsDBImportFileInputsInBatches(int batchSize) throws IOException {
+        final String hg00096 = largeFileTestDir + "gvcfs/HG00096.g.vcf.gz";
+        final String hg00268 = largeFileTestDir + "gvcfs/HG00268.g.vcf.gz";
+        final String na19625 = largeFileTestDir + "gvcfs/NA19625.g.vcf.gz";
+        final String combined = largeFileTestDir + "gvcfs/combined.gatk3.7.g.vcf.gz";
+        final SimpleInterval interval = new SimpleInterval("chr20", 17960187, 17981445);
+
+        testGenomicsDBImporterWithBatchSize(Arrays.asList(hg00096, hg00268, na19625), interval, combined, batchSize);
+    }
+
+    @Test(groups = {"bucket"}, dataProvider = "batchSizes")
+    public void testGenomicsDBImportGCSInputsInBatches(int batchSize) throws IOException {
+        final String hg00096_cloud = getGCPTestInputPath() + "large/gvcfs/HG00096.g.vcf.gz";
+        final String hg00268_cloud = getGCPTestInputPath() + "large/gvcfs/HG00268.g.vcf.gz";
+        final String na19625_cloud = getGCPTestInputPath() + "large/gvcfs/NA19625.g.vcf.gz";
+        final String combined = largeFileTestDir + "gvcfs/combined.gatk3.7.g.vcf.gz";
+        final SimpleInterval interval = new SimpleInterval("chr20", 17960187, 17981445);
+
+        testGenomicsDBImporterWithBatchSize(Arrays.asList(hg00096_cloud, hg00268_cloud, na19625_cloud), interval, combined, batchSize);
+    }
+
+    @Test(expectedExceptions = UserException.class)
+    public void testZeroVCFBufferSize() throws IOException {
+        final String hg00096 = largeFileTestDir + "gvcfs/HG00096.g.vcf.gz";
+        final String hg00268 = largeFileTestDir + "gvcfs/HG00268.g.vcf.gz";
+        final String na19625 = largeFileTestDir + "gvcfs/NA19625.g.vcf.gz";
+        final String combined = largeFileTestDir + "gvcfs/combined.gatk3.7.g.vcf.gz";
+        final SimpleInterval interval = new SimpleInterval("chr20", 17960187, 17981445);
+
+        testGenomicsDBImportWithZeroBufferSize(Arrays.asList(hg00096, hg00268, na19625), interval, combined);
+    }
+
+    private void testGenomicsDBImporter(final List<String> vcfInputs, final SimpleInterval interval, final String expectedCombinedVCF) throws IOException {
+        final String workspace = createTempDir("genomicsdb-tests-").getAbsolutePath() + "/workspace";
+
+        writeToGenomicsDB(vcfInputs, interval, workspace, 0, false, 0);
+        checkJSONFilesAreWritten(workspace);
+        checkGenomicsDBAgainstExpected(workspace, interval, expectedCombinedVCF);
+    }
+
+    private void testGenomicsDBImporterWithBatchSize(final List<String> vcfInputs, final SimpleInterval interval, final String expectedCombinedVCF, int batchSize) throws IOException {
+        final String workspace = createTempDir("genomicsdb-batchsize-tests-").getAbsolutePath() + "/workspace-" + batchSize;
+
+        writeToGenomicsDB(vcfInputs, interval, workspace, batchSize, false, 0);
+        checkGenomicsDBAgainstExpected(workspace, interval, expectedCombinedVCF);
```

Comment:
I found these checks are redundant! GenomicsDB feature reader has lots of checks and if the callset or vidmap files are not generated correctly it fails. Hence, removed